subprojects/core/src/test/groovy/org/gradle/internal/buildevents/BuildResultLoggerTest.groovy

```diff
@@ -42,7 +43,9 @@ class BuildResultLoggerTest extends Specification {
 
         then:
         1 * durationFormatter.format(10L) >> { "10s" }
-        TextUtil.normaliseLineSeparators(textOutputFactory as String) == "{org.gradle.internal.buildevents.BuildResultLogger}{LIFECYCLE}\n{successheader}ACTION SUCCESSFUL{normal} in 10s\n"
+        textOutputFactory.category == BuildResultLogger.canonicalName
+        textOutputFactory.logLevel == LogLevel.LIFECYCLE
+        TextUtil.normaliseLineSeparators(textOutputFactory.output) == "\n{successheader}ACTION SUCCESSFUL{normal} in 10s\n"
```

Comment:
💭 The way you have to specify the styles in the output is also a problem with this fixture.  It almost seems like you'd want to build a simple DSL for easily creating a structured object that encodes the style information and pass that to to the factory to  verify, instead of serializing the info to a string and seeing the serialized form in the test classes.

```
textOutputFactory.assertOutputIs {
  newLine()
  header("ACTION SUCCESSFUL")
  normal(" in 10s")
  newLine()
}
```

This would make the style structure more obvious, I think.  Though it's probably more work than you want to do here now.

Reply:
Please no DSLs in tests.  It can just be a string

Reply:
C'mon, just one little DSL.  Let's go.  Real quick.  In and out, twenty minute DSL.